test/extended/builds/nosrc.go

```diff
@@ -0,0 +1,42 @@
+package builds
+
+import (
+	"fmt"
+
+	g "github.com/onsi/ginkgo"
+	o "github.com/onsi/gomega"
+
+	exutil "github.com/openshift/origin/test/extended/util"
+)
+
+var _ = g.Describe("[builds] build with empty source", func() {
+	defer g.GinkgoRecover()
+	var (
+		buildFixture = exutil.FixturePath("..", "extended", "fixtures", "test-nosrc-build.json")
+		oc           = exutil.NewCLI("cli-build-nosrc", exutil.KubeConfigPath())
+		exampleBuild   = exutil.FixturePath("..", "extended", "fixtures", "test-build-app")
+	)
+
+	g.JustBeforeEach(func() {
+		g.By("waiting for builder service account")
+		err := exutil.WaitForBuilderAccount(oc.KubeREST().ServiceAccounts(oc.Namespace()))
+		o.Expect(err).NotTo(o.HaveOccurred())
+		oc.Run("create").Args("-f", buildFixture).Execute()
+	})
+
+	g.Describe("started build", func() {
+		g.It("should build even with an empty source in build config", func() {
+			g.By("starting the build with --wait flag")
+			out, err := oc.Run("start-build").Args("nosrc-build", "--wait", fmt.Sprintf("--from-dir=%s", exampleBuild)).Output()
```

Comment:
i just fixed a bug in another test related to using the output of start-build to get the name of the build.  i think it's better to hardcode the expected build name, because start-build sometimes returns additional output and then "out" doesn't contain the build name.